src/tokens.js

```diff
@@ -271,101 +272,288 @@ const tokens = {
     // Create token uid
     const tokenUID = this.getTokenUID(input.tx_id, input.index);
-    const authorityOutput = {'address': address, 'value': tokenMasks, 'tokenData': tokenData};
+    const authorityOutput = {'address': address, 'value': tokenMasks, 'tokenData': AUTHORITY_TOKEN_DATA};
+
     // Create tx data
     let txData = {'inputs': [input], 'outputs': [authorityOutput, output], 'tokens': [tokenUID]};
-    // Get data to sign
-    const dataToSign = transaction.dataToSign(txData);
-    // Sign tx
-    txData = transaction.signTx(txData, dataToSign, pin);
-    // Assemble tx and send to backend
-    transaction.completeTx(txData);
-    const txBytes = transaction.txToBytes(txData);
-    const txHex = util.buffer.bufferToHex(txBytes);
     const promise = new Promise((resolve, reject) => {
-      walletApi.sendTokens(txHex, (response) => {
-        if (response.success) {
-          // Save in storage and redux new token configuration
-          this.addToken(response.tx.tokens[0], name, symbol);
-          const mintPromise = this.mintTokens(response.tx.hash, response.tx.tokens[0], address, mintAmount, pin)
-          mintPromise.then(() => {
-            resolve({uid: response.tx.tokens[0], name, symbol});
-          }, (message) => {
-            reject(message);
-          });
-        } else {
-          reject(response.message);
-        }
-      }, (e) => {
-        // Error in request
-        console.log(e);
-        reject(e.message);
+      const txPromise = transaction.sendTransaction(txData, pin);
+      txPromise.then((response) => {
+        // Save in storage and redux new token configuration
+        this.addToken(response.tx.tokens[0], name, symbol);
+        const mintPromise = this.mintTokens(response.tx.hash, 0, address, response.tx.tokens[0], address, mintAmount, pin, true, true);
+        mintPromise.then(() => {
+          resolve({uid: response.tx.tokens[0], name, symbol});
+        }, (message) => {
+          reject(message);
+        });
+      }, (message) => {
+        reject(message);
       });
     });
     return promise;
   },
 
+  /**
+   * Generate mint data
+   *
+   * @param {string} txID Hash of the transaction to be used to mint tokens
+   * @param {number} index Index of the output being spent
+   * @param {string} addressSpent Address of the output being spent
+   * @param {string} token Token uid to be minted
+   * @param {string} address Address to receive the amount of the generated token
+   * @param {number} amount Amount of the token that will be minted
+   * @param {boolean} createAnotherMint If should create another mint output after spending this one
+   * @param {boolean} createMelt If should create a melt output (useful when creating a new token)
+   *
+   * @return {Object} Mint data {'inputs', 'outputs', 'tokens'}
+   *
+   * @memberof Tokens
+   * @inner
+   */
+  getMintData(txID, index, addressSpent, token, address, amount, createAnotherMint, createMelt) {
```

Comment:
I would rename it to `createMintData` (instead of get).

Reply:
Done